auth/auth.go

```diff
@@ -206,11 +225,15 @@ func NewCredentials(opts *CredentialsOptions) *Credentials {
 // CachedTokenProvider.
 type CachedTokenProviderOptions struct {
 	// DisableAutoRefresh makes the TokenProvider always return the same token,
-	// even if it is expired.
+	// even if it is expired. The default is false. Optional.
 	DisableAutoRefresh bool
 	// ExpireEarly configures the amount of time before a token expires, that it
-	// should be refreshed. If unset, the default value is 10 seconds.
+	// should be refreshed. If unset, the default value is 3 minutes and 45
+	// seconds. Optional.
 	ExpireEarly time.Duration
+	// BlockingRefresh configures a synchronous workflow that refreshes
+	// stale tokens while blocking. The default is false. Optional.
+	BlockingRefresh bool
```

Comment:
nit: I wonder if this should be DisableAysyncRefresh for parity with the other bool option on this struct. Thoughts?

Reply:
It might be good for the options to be more different than alike, in order to avoid user bugs in which someone mistakenly types or copies the wrong option due to the similarity. `DisableAutoRefresh` and `DisableAsyncRefresh` are really visually similar. But if you think the parity is the better call, I'm more than happy to make the change.

Reply:
I think I like the parity as the surface presents a clearer design choice for boolean options. I believe we have tried to use enabled disabled for most of our newer bool options.

Reply:
SGTM, will do.

Reply:
Done, thanks!